tests/jax/components/executing/observing_test.py

```diff
@@ -0,0 +1,269 @@
+# python3
+# Copyright 2021 InstaDeep Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for FeedforwardExecutorObserve class for Jax-based Mava systems"""
+
+
+from types import SimpleNamespace
+from typing import Dict
+
+import pytest
+from acme.types import NestedArray
+from dm_env import StepType, TimeStep
+
+from mava.components.jax.executing.observing import (
+    ExecutorObserveConfig,
+    FeedforwardExecutorObserve,
+)
+from mava.systems.jax.executor import Executor
+from mava.types import OLT
+from mava.utils.sort_utils import sort_str_num
+
+
+class MockAdder:
+    def __init__(self):
+        self.parm = "empty"
+
+    def add_first(
+        self, timestep: TimeStep, extras: Dict[str, NestedArray] = ...
+    ) -> None:
+        self.parm = "after_add_first"
+
+    def add(
+        self,
+        actions: Dict[str, NestedArray],
+        next_timestep: TimeStep,
+        next_extras: Dict[str, NestedArray] = ...,
+    ) -> None:
+        self.parm = "after_add"
+
+
+class MockExecutorParameterClient:
+    def __init__(self):
+        self.parm = False
+
+    def get_async(self):
+        self.parm = True
+
+
+@pytest.fixture
+def mock_executor_without_adder() -> Executor:
+    """Mock executor component without adder"""
+    store = SimpleNamespace(is_evaluator=None, observations={}, adder=None)
+    return Executor(store=store)
+
+
+@pytest.fixture
+def mock_executor_without_parameter_client() -> Executor:
+    """Mock executor component without parameter_client"""
+    store = SimpleNamespace(
+        is_evaluator=None, observations={}, executor_parameter_client=None
+    )
+    return Executor(store=store)
+
+
+class MockExecutor(Executor):
+    def __init__(self, *args: object) -> None:
+        # agent_net_keys
+        agent_net_keys = {
+            "agent_0": "network_agent_0",
+            "agent_1": "network_agent_1",
+            "agent_2": "network_agent_2",
+        }
+        # network_int_keys_extras
+        network_int_keys_extras = None
+        # network_sampling_setup
+        network_sampling_setup = [
```

Comment:
I think it is better in testing to have to values of variables set explicitly as much as possible instead of having them computed. That way we know exactly what we are working with. In this case this variable can then be set as `network_sampling_setup = [["network_agent_0", "network_agent_1", "network_agent_2"]]`